packages/flutter/test/widgets/framework_test.dart

```diff
@@ -1427,6 +1427,64 @@ void main() {
       expect(debugDoingBuildOnDidUnmountRenderObject, isFalse);
     });
   });
+
+  testWidgets('Error messages', (WidgetTester tester) async {
+    final GlobalKey key = GlobalKey();
+    await tester.pumpWidget(Container(child: _WidgetWithNoVisitChildren(_StatefulLeaf(key: key))));
+    (key.currentState as _StatefulLeafState).markNeedsBuild();
+    await tester.pumpWidget(Container());
+    final dynamic exception = tester.takeException();
```

Comment:
I am surprised this is not allowed. What if there is an inherited widget that both container and _StatefulLeaf depend on. When the inherited widget mark both them dirty, the container rebuilt without the _StatefulLeaf?

Reply:
I'm not sure what you think should be allowed? Can you elaborate?

My intent was for this to test the error message that is reported as a result of a bad `visitChildren`.

Reply:
Oh Nvm, I was confused at the setup, I think this is fine. We should update the test title to be a bit more clear though.